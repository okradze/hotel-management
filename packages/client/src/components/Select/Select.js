import React, { useState } from 'react'
import OutsideClickHandler from 'react-outside-click-handler'

const Select = ({ setValue, options, value }) => {
    const [isOpen, setIsOpen] = useState(false)

    return (
        <div className="select">
            <OutsideClickHandler onOutsideClick={() => setIsOpen(false)}>
                <button
                    onClick={() => setIsOpen(!isOpen)}
                    type="button"
                    className="select__button"
                >
                    <span>{value}</span>
                </button>
                {isOpen && (
                    <div className="select__list-wrapper">
                        <ul className="select__list">
                            {options.map(({ text, value }, index) => (
                                <li key={index} className="select__item">
                                    <button
                                        onKeyPress={e => {
                                            if (e.which === 13) {
                                                setValue({ ...value, text })
                                                setIsOpen(false)
                                            }
                                        }}
                                        onClick={() => {
                                            setValue({ ...value, text })
                                            setIsOpen(false)
                                        }}
                                        className="button--transparent"
                                        type="button"
                                    >
                                        {text}
                                    </button>
                                </li>
                            ))}
                        </ul>
                    </div>
                )}
            </OutsideClickHandler>
        </div>
    )
}

export default Select
