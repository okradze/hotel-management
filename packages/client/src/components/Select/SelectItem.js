import React from 'react'

const SelectItem = ({ setIsOpen, setValue, text, value }) => (
    <li className="select__item">
        <button
            onKeyPress={e => {
                if (e.which === 13) {
                    setValue(text, value)
                    setIsOpen(false)
                }
            }}
            onClick={() => {
                setValue(text, value)
                setIsOpen(false)
            }}
            className="button--transparent"
            type="button"
        >
            {text}
        </button>
    </li>
)

export default SelectItem
