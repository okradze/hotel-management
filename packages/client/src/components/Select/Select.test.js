import Select from './Select'

describe('Select', () => {
    let wrapper

    const setIsOpen = jest.fn()
    const useStateSpy = jest.spyOn(React, 'useState')
    useStateSpy.mockImplementation(init => [init, setIsOpen])

    const options = [
        {
            text: '1',
            value: '1',
        },
        {
            text: '2',
            value: '2',
        },
    ]

    beforeEach(() => {
        wrapper = shallow(
            <Select
                value={options[0].text}
                options={options}
                setValue={(text, value) => setValue(text, value)}
            />,
        )
    })

    it('renders when closed', () => {
        expect(wrapper).toMatchSnapshot()
    })
    it('renders when is open', () => {
        wrapper.find('button').simulate('click')

        expect(setIsOpen).toHaveBeenCalledWith(true)
        expect(wrapper).toMatchSnapshot()
    })
})
