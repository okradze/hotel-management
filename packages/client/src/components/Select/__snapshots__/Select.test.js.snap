// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders when closed 1`] = `
<div
  className="select"
>
  <OutsideClickHandler
    disabled={false}
    display="block"
    onOutsideClick={[Function]}
    useCapture={true}
  >
    <button
      className="select__button"
      onClick={[Function]}
      type="button"
    >
      <span>
        1
      </span>
    </button>
  </OutsideClickHandler>
</div>
`;

exports[`Select renders when is open 1`] = `
<div
  className="select"
>
  <OutsideClickHandler
    disabled={false}
    display="block"
    onOutsideClick={[Function]}
    useCapture={true}
  >
    <button
      className="select__button"
      onClick={[Function]}
      type="button"
    >
      <span>
        1
      </span>
    </button>
  </OutsideClickHandler>
</div>
`;
